Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -276,6 +280,30 @@ def adaptationNoteLinter : TextbasedLinter := fun lines ↦ Id.run do
     lineNumber := lineNumber + 1
   return errors
 
+/-- Strip leading and trailing spaces. -/
+def stripSpaces (s : String) : String := Id.run do
+  let mut stripped := s
+  while stripped.startsWith " " do
+    stripped := stripped.stripPrefix " "
+  while stripped.endsWith " " do
+    stripped := stripped.stripSuffix " "
+  return stripped
+
+/-- Lint on a collection of input strings if one of the is a duplicate import statement. -/
+def duplicateImportsLinter : TextbasedLinter := fun lines ↦ Id.run do
+  let mut lineNumber := 1
+  let mut errors := Array.mkEmpty 0
+  let mut importStatements := Array.mkEmpty 0
+  for line in lines do
+    if line.startsWith "import " then
+      let lineWithoutComment := (line.splitOn "--")[0]!
+      let mut strippedLine := stripSpaces lineWithoutComment
+        if importStatements.contains strippedLine then
```

Comment:
This should work, but is quadratic in the number of imports. (In practice, that should not matter much.)
A slightly more efficient way would be to sort the imports at the end, and then check if any two consecutive imports are the same (using `List.eraseReps` or similar).